tests/settings_tests/tests.py

```diff
@@ -441,3 +441,132 @@ def test_tuple_settings(self):
+    """
+
+    def setUp(self):
+        signals.setting_changed.connect(self.receiver)
+        self.addCleanup(signals.setting_changed.disconnect, self.receiver)
+
+        # Create a spy that will be connected to `setting_changed` signal
+        # and executed AFTER `self.receiver`.
+        self.spy_receiver = mock.Mock()
+        signals.setting_changed.connect(self.spy_receiver)
+        self.addCleanup(signals.setting_changed.disconnect, self.spy_receiver)
+
+    def receiver(self, **kwargs):
+        """
+        A receiver for demonstration purposes.
+        It fails while certain settings are being changed.
+
+        - BUG_29467_BOTH raises an error while receiving the signal
+          on both entering and exiting the context manager
+        - BUG_29467_ENTER raises an error only on enter.
+        - BUG_29467_EXIT raises an error only on exit.
+        """
+        setting = kwargs['setting']
+        enter = kwargs['enter']
+
+        if setting in ('BUG_29467_BOTH', 'BUG_29467_ENTER') and enter:
+            raise SettingChangeEnterException
+        if setting in ('BUG_29467_BOTH', 'BUG_29467_EXIT') and not enter:
+            raise SettingChangeExitException
+
+    def check_settings(self):
+        """
+        Ensure that settings for this test case are missing from the settings module.
+        """
+        self.assertFalse(hasattr(settings, 'BUG_29467_BOTH'))
+        self.assertFalse(hasattr(settings, 'BUG_29467_ENTER'))
+        self.assertFalse(hasattr(settings, 'BUG_29467_EXIT'))
+        self.assertFalse(hasattr(settings, 'BUG_29467_PASS'))
+
+    def check_spy_receiver_exit_calls(self, call_count):
+        """
+        Ensure that `self.spy_receiver` was called exactly `call_count` times
+        with ``enter=False`` keyword argument.
+
+        :type call_count: int
+        """
+        kwargs_with_exit = [kwargs for args, kwargs in self.spy_receiver.call_args_list
+                            if ('enter', False) in kwargs.items()]
+
+        self.assertEqual(len(kwargs_with_exit), call_count)
+
+    def test_override_settings_both(self):
+        """
+        Receiver fails on both enter and exit.
+        """
+        with self.assertRaises(SettingChangeEnterException):
+            with override_settings(BUG_29467_PASS='BOTH', BUG_29467_BOTH='BOTH'):
+                pass
+
+        self.check_settings()
+
+        # We touched two settings, so we expect two calls of `spy_receiver`.
+        self.check_spy_receiver_exit_calls(call_count=2)
+
+    def test_override_settings_enter(self):
+        """
+        Receiver fails on enter only.
+        """
+        with self.assertRaises(SettingChangeEnterException):
+            with override_settings(BUG_29467_PASS='ENTER', BUG_29467_ENTER='ENTER'):
+                pass
+
+        self.check_settings()
+
+        # We touched two settings, so we expect two calls of `spy_receiver`.
```

Comment:
Avoid use of "we" in comments per our [style guide](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).